src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -656,6 +656,13 @@ JNI_METHOD(void, pairDevice)
         JniByteArray jniCsrNonce(env, csrNonce);
         commissioningParams.SetCSRNonce(jniCsrNonce.byteSpan());
     }
+
+    commissioningParams.SetICDRegistrationStrategy(ICDRegistrationStrategy::kBeforeComplete);
```

Comment:
this is not right place to put commissioningParams.SetICDRegistrationStrategy(ICDRegistrationStrategy::kBeforeComplete);
Comment

Reply:
offline discussion, it should be fine here.